framework/include/vectorpostprocessors/VectorPostprocessorData.h

```diff
@@ -64,26 +70,27 @@ class VectorPostprocessorData : public Restartable
   /**
    * Get the map of names -> VectorPostprocessor values. Exposed for error checking.
    */
-  const std::map<std::string, std::map<std::string, VectorPostprocessorValue*> > & values() const { return _values; }
+//  const std::map<std::string, std::map<std::string, VectorPostprocessorValue*> > & values() const { return _values; }
 
   /**
    * Get the map of vectors for a particular VectorPostprocessor
    * @param vpp_name The name of the VectorPostprocessor
    */
-  const std::map<std::string, VectorPostprocessorValue*> & vectors(const std::string & vpp_name) { return _values[vpp_name]; }
+  const std::map<std::string, VPPVectors> & vectors(const std::string & vpp_name) const;
 
   /**
    * Copy the current post-processor values into old (i.e. shift it "back in time")
    */
   void copyValuesBack();
 
-protected:
+private:
+  VectorPostprocessorValue & getVectorPostprocessorHelper(const VectorPostprocessorName & vpp_name, const std::string & vector_name, bool get_current);
 
-  /// Values of the post-processor at the current time
-  std::map<std::string, std::map<std::string, VectorPostprocessorValue*> > _values;
+  /// Values of the vector post-processor
+  std::map<std::string, std::map<std::string, VPPVectors> > _values;
 
-  /// Values of the post-processors at the time t-1
-  std::map<std::string, std::map<std::string, VectorPostprocessorValue*> > _values_old;
+  std::set<std::string> _requested_items;
+  std::set<std::string> _supplied_items;
 };
```

Comment:
Yes and no. Right now there's already dependency resolution on the objects themselves because they are UserObjects and sorted in the warehouse. The intention for this was to add in a new sanity check for the vectors themselves if you couple to vectors, there is no check on whether a vector that you couple to exists right now. I started on this but need to flesh it out quite a bit more and want to unify the way it works between VectorPostprocessors and Postprocessors. I can either continue to work on this or continue that work in another PR. I kind of want to handle it separately since this is already getting to be a substantial PR.